From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 29 Apr 2019 11:28:45 +0200
Subject: [PATCH] Don't close fds when executing programs

This is basically the same as ca894a24fca452e5aa982e1fc5dc63117c3e3d71 just
for new new Popen_safe_legacy().
---
 mesonbuild/mesonlib.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/mesonlib.py b/mesonbuild/mesonlib.py
index 98c2366d6a48..b5df8473ee40 100644
--- a/mesonbuild/mesonlib.py
+++ b/mesonbuild/mesonlib.py
@@ -917,7 +917,7 @@ def Popen_safe(args, write=None, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
     return p, o, e
 
 def Popen_safe_legacy(args, write=None, stdout=subprocess.PIPE, stderr=subprocess.PIPE, **kwargs):
-    p = subprocess.Popen(args, universal_newlines=False,
+    p = subprocess.Popen(args, universal_newlines=False, close_fds=False,
                          stdout=stdout, stderr=stderr, **kwargs)
     if write is not None:
         write = write.encode('utf-8')
